server/api/routes/users.py

```diff
@@ -1,42 +1,86 @@
 from flask import Blueprint, request
 from flask_jwt_extended import create_access_token
 from database import User
+from api.routes.validators.users import valid_email, valid_password
 
 users = Blueprint('users', __name__)
 
-
 @users.route('/login', methods=['POST'])
 def login():
     """POST users/login
 
     Desc: Logs in a user
+          Ensures fields are not blank
+          Validates email used to log in
+          Checks if an account exists at all
+          Checks if an account exists and the password was incorrect
 
     Returns:
         200: success - true
-        401: success - false
+        401: password was incorrect, unauthorized access
+        404: no account was found
     """
     req = request.get_json()
-    user = User.objects(email=req['email'], password=req['password']).first()
-    if user:
-        access_token = create_access_token(identity=user)
+    if req['email'] == '' or req['password'] == '':
```

Comment:
Can this be abstracted into the validator? I thought the validator functions already checked if the fields were empty.

Reply:
The email one can but I didn't want to validate passwords during login, haven't seen that done anywhere. That would be like if a user entered an incorrect password and the error message they got was 'passwords must contain a number' instead of 'password was incorrect'.

Reply:
Well not that necessarily. I more just meant that have a login validator that returns some message saying "password cannot be empty" or something. Just to contain all validation logic separately from the route itself

Reply:
Ah gotcha, can do that